### ISSUE TITLE
L2 Regularization

### ISSUE THREAD

**Issue by axiomizer** [axiomizer/haida-go#1]
![image](https://github.com/axiomizer/haida-go/assets/138620999/3b2817e7-7ab3-4fd5-9298-51facc584919)

The "Methods -> Optimization" section mentions that c = 10^-4

[Mastering the game of Go without human knowledge](https://www.nature.com/articles/nature24270.epdf?author_access_token=VJXbVjaSHxFoctQQ4p2k4tRgN0jAjWel9jnR3ZoTv0PVW4gB86EEpGqTRDtpIz-2rmo8-KG06gqVobU5NSCFeHILHcVFUeMsbvwS-lxjqQGg98faovwjxeTUgZAUMnRQ)

**Comment by axiomizer** [comment]
Implemented in 80ec767266f050f3e39da453b14033ba978b6c5a.

In the unit tests, I'm passing in weight_decay to torch.optim.SGD, but even if nothing is passed in, the test won't fail. This is because the weight decay is so small. If you change it to 0.1, it has enough of an effect to make the test fail unless you pass it into torch.optim.SGD.

**Comment by axiomizer** [comment]
Incorporate weight_decay into test_full_net with commits 69a282b11a443adb6d63cbe06b354e50b73fc4d4 and a49326de28b4f47a8a3339220d82b6acb4c4bcd5

**Comment by axiomizer** [comment]
Ended up writing a weight decay test, included incidentally in commit 270b8c424a60984d6e9ef706f50c4efe79cbd085 and improved in commit 773db595507d82c58191d4e43c5edb10daf2f22e
